### PR TITLE
Update O3DE to use latest gamelift server sdk

### PR DESCRIPTION
**Pull request by onecent1101**
## Testing
Pass unit test
Tested locally with GameLiftLocal

Signed-off-by: onecent1101 <liug@amazon.com>

### CODE REVIEW

File: cmake/3rdParty/Platform/Windows/BuiltInPackages_windows.cmake [hunk 1]
```diff
@@ -19,8 +19,8 @@ ly_associate_package(PACKAGE_NAME xxhash-0.7.4-rev1-multiplatform
 
 # platform-specific:
 ly_associate_package(PACKAGE_NAME assimp-5.1.6-rev1-windows                             TARGETS assimplib                   PACKAGE_HASH 299d8a3c70657d74af8841650538e9d083fda9356f6782416edbec0ef5a0493e)
-ly_associate_package(PACKAGE_NAME OpenEXR-3.1.3-rev2-windows	                        TARGETS OpenEXR                     PACKAGE_HASH cf7707193d4b1c3dd9de7a940641d7750522fc1079f47fde145f31052206a6f3)
-ly_associate_package(PACKAGE_NAME AWSGameLiftServerSDK-3.4.1-rev1-windows               TARGETS AWSGameLiftServerSDK        PACKAGE_HASH a0586b006e4def65cc25f388de17dc475e417dc1e6f9d96749777c88aa8271b0)
+ly_associate_package(PACKAGE_NAME OpenEXR-3.1.3-rev2-windows                            TARGETS OpenEXR                     PACKAGE_HASH cf7707193d4b1c3dd9de7a940641d7750522fc1079f47fde145f31052206a6f3)
```
Comment (allisaurus) [inline cmake/3rdParty/Platform/Windows/BuiltInPackages_windows.cmake:22]: nit: extra tab after pkg name, maybe? does not look like this line has meaningfully changed

**Review by allisaurus — APPROVED** [review]
LG save one small whitespace nit

File: cmake/3rdParty/Platform/Windows/BuiltInPackages_windows.cmake (hunk 1 above)
Comment (onecent1101) [reply cmake/3rdParty/Platform/Windows/BuiltInPackages_windows.cmake:22]: actually previously it was a tab, I am fixing it to space

**Review by lmbr-pip — APPROVED** [review]
LG

**Review by junbo75 — APPROVED** [review]

**Pull request merged**